datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -607,6 +609,11 @@ def _submit_metric(
         if hostname is None:
             hostname = ''
 
+        if self.global_metrics_filter is not None:
+            collect_metric = re.match(self.global_metrics_filter, name)
```

Comment:
Would it make sense to pre-compile the regular expression to make matching faster?

Reply:
True, thanks for the suggestion!